code/modules/clothing/modular_armor/style_line.dm

```diff
@@ -80,3 +80,274 @@
 		"gold",
 	)
 	current_variant = "black"
+
+// ***************************************
+//  Modular hats
+/obj/item/clothing/head/modular/style
+	name = "\improper Nice Hat"
+	desc = "Nice hat bro. How did you find this?"
+	flags_item_map_variant = NONE
+	icon = 'icons/mob/modular/style_hats.dmi'
+	item_icons = list(
+		slot_head_str = 'icons/mob/modular/style_hats_mob.dmi',
+	)
+	attachments_allowed = list(
+		/obj/item/armor_module/greyscale/badge,
+		/obj/item/armor_module/storage/helmet,
+	)
+
+	flags_inv_hide = NONE
+
+	soft_armor = list(MELEE = 15, BULLET = 15, LASER = 15, ENERGY = 15, BOMB = 10, BIO = 5, FIRE = 5, ACID = 5)
+	starting_attachments = list(/obj/item/armor_module/storage/helmet)
+
+/obj/item/clothing/head/modular/style/update_item_sprites()
+	switch(SSmapping.configs[GROUND_MAP].armor_style)
+		if(MAP_ARMOR_STYLE_JUNGLE)
+			if(flags_item_map_variant & ITEM_JUNGLE_VARIANT)
+				current_variant = "drab"
+		if(MAP_ARMOR_STYLE_ICE)
+			if(flags_item_map_variant & ITEM_ICE_VARIANT)
+				current_variant = "snow"
+		if(MAP_ARMOR_STYLE_PRISON)
+			if(flags_item_map_variant & ITEM_PRISON_VARIANT)
+				current_variant = "black"
+		if(MAP_ARMOR_STYLE_DESERT)
+			if(flags_item_map_variant & ITEM_DESERT_VARIANT)
+				current_variant = "desert"
+
+
+//marine hats
+/obj/item/clothing/head/modular/style/beret
+	name = "TGMC beret"
+	desc = "A hat used by the TGMC, typically considered the most iconic military headgear. Often reserved for higher ranking officers, they occasionally they find their way down the ranks into the hands of squad-leaders and decorated grunts."
+	icon_state = "tgmc_beret"
+	flags_item_map_variant = ITEM_JUNGLE_VARIANT|ITEM_ICE_VARIANT|ITEM_DESERT_VARIANT
+
+	icon_state_variants = list(
+		"drab",
+		"snow",
+		"black",
+		"desert",
+		"red",
+		"blue",
+		"purple",
+		"gold",
+	)
+
+	current_variant = "black"
```

Comment:
just set this at the parent level instead of every child type.

Reply:
Fair enough.